homeassistant/components/camera/__init__.py

```diff
@@ -129,20 +132,21 @@ def _proxy_camera_mjpeg_stream(handler, path_match, data):
             while True:
 
                 img_bytes = camera.camera_image()
-
-                headers_str = '\r\n'.join((
-                    'Content-length: {}'.format(len(img_bytes)),
-                    'Content-type: image/jpeg',
-                )) + '\r\n\r\n'
-
-                handler.request.sendall(
-                    bytes(headers_str, 'utf-8') +
-                    img_bytes +
-                    bytes('\r\n', 'utf-8'))
-
-                handler.request.sendall(
-                    bytes('--jpgboundary\r\n', 'utf-8'))
-
+                if img_bytes is not None:
```

Comment:
Instead of breaking, I think we should make it just skip a frame. Add this to line 135.

```python
if img_bytes is None:
    continue
```